### ISSUE TITLE
1000 soil - process metaG data: JGI & Genewiz

### ISSUE THREAD

**Issue by mslarae13** [microbiomedata/issues#613]
1000 soils has  metaG data generated from JGI and from Genewiz

16 : # of samples run at JGI : See https://github.com/microbiomedata/issues/issues/85 
- [x] Map JGI samples to NMDC biosamples
- [x] Update NMDC metadata or JGI depending on which is right

?? : # of samples run at Genewiz
- [x] Get data on NERSC to process (in progress)
- [x] Map data files to NMDC biosamples (Bea, in Pprogress)
- [x] https://github.com/microbiomedata/issues/issues/632
- [ ] Process All data

running binning and MAGs for JGI results on JGI assemblies is blocked on https://github.com/microbiomedata/nmdc_automation/issues/35

**Comment by ssarrafan** [comment]
Appears to be in progress and active. I'll move this to the next sprint @bmeluch @mslarae13

**Comment by bmeluch** [comment]
@JamesTessmer @sujaypatil96 here is the issue for tracking work on the 1000 soils metagenomics OmicsProcessing records and getting the data into workflows, etc. Thanks so much for working on this!

**Comment by aclum** [comment]
@bmeluch  @lamccue Do we know anything about the instrument model for the data from Genewiz? Its not required by the schema but it is a data portal faceted search option so would be nice to have this for the omics records.

**Comment by lamccue** [comment]
I'm not entirely sure.  
Illumina definitely.
More specifically: _maybe_ Illumina NovaSeq

**Comment by mslarae13** [comment]
Discussed in a quick meeting

```
"omics_type": {
        "has_raw_value": "Metagenome"

"part_of": ["nmdc:sty-11-28tm5d36"]

"type": "nmdc:OmicsProcessing"
name == biosample name (edited) 

"instrument_name": "Illumina NovaSeq S4"

"principal_investigator": {
        "email": "nancy.hess@pnnl.gov",
        "has_raw_value": "Nancy Hess",
        "name": "Nancy Hess" 

```

name == biosample name

```
id: nmdc:omprc-11-123
has_input:
 - nmdc:bsm-11-yj9yav68
has_output:
 - nmdc:dobj-11-123
 - nmdc:dobj-11-124```

**Comment by mslarae13** [comment]
`processing_institution` will be left empty until this issue us resolved: https://github.com/microbiomedata/issues/issues/634

**Comment by mslarae13** [comment]
@sujaypatil96 & @JamesTessmer have created the -omics processing records. 
Workflows should be ready to be run for both the JGI and Genewiz (Azenta) data.

@aclum , @mbthornton-lbl do you need anything else?

**Comment by aclum** [comment]
No we should be good to go. Michael has already started on getting the JGI data staged.

**Comment by ssarrafan** [comment]
@mslarae13 @bmeluch can this issue be closed based on Alicia's last comment?

**Comment by bmeluch** [comment]
I'm not sure, would have to ask @aclum or @mbthornton-lbl how the workflows stuff is progressing

**Comment by aclum** [comment]
This needs to go in the next sprint for the data processing.

**Comment by aclum** [comment]
@JamesTessmer Would you pleae issue an changesheet for the omics records you made to change 'omics_type.has_raw_value':'metagenome' to 'omics_type.has_raw_value':'Metagenome'

**Comment by JamesTessmer** [comment]
@aclum I'm working on the change sheet, but I've run into some formatting issues I think.
When I try to validate the change sheet I get an error saying "'Action' column is missing" but it's there in the file.
It won't let me attach a TSV file here so I pasted the first 3 lines, and it seems like it might just be a formatting issue I can't figure out. All the content in the file should be correct though. Sorry if this is some easy error to fix, I haven't used the change sheets before.

```
id	action	attribute	value
nmdc:omprc-11-5r7kbf66    update    omics_type.has_raw_value    Metagenome
nmdc:omprc-11-46tjy763    update    omics_type.has_raw_value    Metagenome
```

**Comment by JamesTessmer** [comment]
> @aclum I'm working on the change sheet, but I've run into some formatting issues I think. When I try to validate the change sheet I get an error saying "'Action' column is missing" but it's there in the file. It won't let me attach a TSV file here so I pasted the first 3 lines, and it seems like it might just be a formatting issue I can't figure out. All the content in the file should be correct though. Sorry if this is some easy error to fix, I haven't used the change sheets before.
> 
> ```
> id	action	attribute	value
> nmdc:omprc-11-5r7kbf66    update    omics_type.has_raw_value    Metagenome
> nmdc:omprc-11-46tjy763    update    omics_type.has_raw_value    Metagenome
> ```

Ah, of course shortly after posting this I got it figured out. There was a space pretending to be a tab. The change sheet passes validation now.

**Comment by scanon** [comment]
Genewiz generated data is being processed now.

**Comment by sujaypatil96** [comment]
Just wanted to make a quick note here that the JGI data objects do not have paths on the "url" slot asserted on them yet. Not sure if that information is required by the `nmdc_automation` code?

**Comment by sujaypatil96** [comment]
If it is, I can get the paths/URLs from @mflynn-lanl and add that to the data objects. If not, we should still add them, looking at it from a "metadata enrichment" lens.

**Comment by aclum** [comment]
@sujaypatil96 we should talk about this more, JGI requires login for download so i'm not sure we want to use the same slot @turbomam

**Comment by mflynn-lanl** [comment]
The JGI samples are also stored on tape and they need to be restored before they can be downloaded

**Comment by aclum** [comment]
Actively in progress. All genewiz data has been processed through annotation. Binning still needs to be run. Ingest of JGI data is still outstanding. Moving to the next sprint.

**Comment by aclum** [comment]
backlogging this until re-iding is done.

**Comment by mbthornton-lbl** [comment]
Query:
```
db.getCollection("data_generation_set").find({
    "associated_studies": "nmdc:sty-11-28tm5d36",
    "analyte_category": "metagenome"
}, {"id": 1, "_id": 0}
)
``` 


Found IDs:
<details><summary>Details</summary>
<p>

nmdc:omprc-12-zkpw2e46
nmdc:omprc-12-y3atvn08
nmdc:omprc-12-scg48547
nmdc:omprc-12-qhne3f15
nmdc:omprc-12-pk2jn108
nmdc:omprc-12-nv67tj64
nmdc:omprc-12-nct8f131
nmdc:omprc-12-mm525w16
nmdc:omprc-12-hgksne68
nmdc:omprc-12-h5aayv46
nmdc:omprc-12-c0nn7528
nmdc:omprc-12-c06jgr44
nmdc:omprc-12-bk872674
nmdc:omprc-12-bcd5ve19
nmdc:omprc-12-1gf6f154
nmdc:omprc-11-z89xbm50
nmdc:omprc-11-xznvyc57
nmdc:omprc-11-xpsvv759
nmdc:omprc-11-xe45df73
nmdc:omprc-11-whasw455
nmdc:omprc-11-w6ny1n56
nmdc:omprc-11-vtzfjf76
nmdc:omprc-11-tf5be523
nmdc:omprc-11-t57e9b93
nmdc:omprc-11-t428ve46
nmdc:omprc-11-sh5mve61
nmdc:omprc-11-s9ycgy76
nmdc:omprc-11-rj54z444
nmdc:omprc-11-re79te58
nmdc:omprc-11-r6hm2m82
nmdc:omprc-11-qe9bj064
nmdc:omprc-11-n6sn5106
nmdc:omprc-11-kewtc376
nmdc:omprc-11-h311fd14
nmdc:omprc-11-h2z4kv61
nmdc:omprc-11-ga0caw33
nmdc:omprc-11-fmxcdq73
nmdc:omprc-11-evy3qc41
nmdc:omprc-11-bfset069
nmdc:omprc-11-av1kpg17
nmdc:omprc-11-amy51k67
nmdc:omprc-11-9qzjk067
nmdc:omprc-11-982chk72
nmdc:omprc-11-8sjf2k26
nmdc:omprc-11-8chjy581
nmdc:omprc-11-7tjt5a93
nmdc:omprc-11-5r7kbf66
nmdc:omprc-11-54xrpm24
nmdc:omprc-11-4527pn32
nmdc:omprc-11-2k9n2g07


</p>
</details>

**Comment by mbthornton-lbl** [comment]
Added to Prod Scheduler - jobs scheduling and being picked up by the watcher

**Comment by aclum** [comment]
I think this is the projects where the jobs are queued but not scheduling. Needs debugging.

**Comment by aclum** [comment]
Some of these ran & others did not. @kaijli can you check on this. The above list from @mbthornton-lbl from two weeks ago appears to be truncated. 

query for outstanding projects


```
db.getCollection('data_generation_set').aggregate(
--
[
{
$match: {
analyte_category: 'metagenome',
processing_institution: 'JGI',
associated_studies: 'nmdc:sty-11-28tm5d36'
}
},
{
$lookup: {
from: 'workflow_execution_set',
localField: 'id',
foreignField: 'was_informed_by',
as: 'workflow_execution_set'
}
},
{
$match: {
workflow_execution_set: { $size: 2 }
}
},
{ $project: { id: 1 } }
],
{ maxTimeMS: 60000, allowDiskUse: true }
);

```


[20250320_oustanding_1000 soils.csv](https://github.com/user-attachments/files/19376662/20250320_oustanding_1000.soils.csv)

**Comment by aclum** [comment]
moving to the next sprint.

**Comment by kaijli** [comment]
Submitted to prod allow.lst

**Comment by ssarrafan** [comment]
@aclum @kaijli is there more to do here?

**Comment by aclum** [comment]
Jobs were behind NEON surface samples so haven't run yet, moving to the next sprint.

**Comment by kaijli** [comment]
Resubmitted, IDs listed on rancher in `/conf/submit_to_scheduler/1000_soil_t613` (new folder to collect different IDs for concatenating into one `allow.lst` for submission)

**Comment by aclum** [comment]
92/101 metagenomes processed.

**Comment by kaijli** [comment]
Released the jobs via the api endpoint for annotation v1.1.5 and rba v1.0.10 and they've resubmitted on JAWS.

**Comment by ssarrafan** [comment]
Moving to new sprint.

**Comment by aclum** [comment]
We have mags analysis for all 101 of these metagenomes now.
